src/Online_Bookstore/CustomHooks/useFetch.js

```diff
@@ -0,0 +1,22 @@
+import {useCallback, useEffect,useState} from 'react'
+import axios from 'axios'
+ const useFetch = (url) => {
+  const[booksData,setBookData]=useState([])
+  const getBooksData=useCallback(async()=>{
+    try{
+    const response=await axios.get(url);
+    const {data} = await response;
+    setBookData(data);
+    
+    }catch(err)
+    {
+      console.log(err);
```

Comment:
Instead of logging the error to the console, follow a proper approach to handling errors. You can either 'Throw the Error' with a valid message. Or set a state. Refer https://linkpe.in/story/react-fetch-data. This is just a single example, you can search for more.

@Poorvi-3pillar